framework/Metrics/metrics/StatsTestMetric.py

```diff
@@ -0,0 +1,97 @@
+from Metrics.metrics import MetricUtilities
+from utils import InputData
+import StatsTestUtils as sU
+
+class StatsTestMetric(MetricInterface):
+  """
+    Metric to compare two datasets using statistical tests.
+  """
+  available_tests = {}
+  available_tests['full']['f_test'] = stats.f_oneway
+  available_tests['full']['chi_square'] = stats.chi_square
+  available_tests['full']['ks_test'] = stats.ks_2samp
+  available_tests['seg']['f_test'] = sU.compare_segments_f
+  available_tests['seg']['chi_square'] = sU.compare_segments_chi
+  available_tests['seg']['ks_test'] = sU.compare_segments_ks
+  available_tests['int']['f_test'] = sU.compare_interval_f
+  available_tests['int']['chi_square'] = sU.compare_interval_chi
+  available_tests['int']['ks_test'] = sU.compare_interval_ks
+  available_tests['seg_sum']['f_test'] = sU.compare_sums_f
+  available_tests['seg_sum']['chi_square'] = sU.compare_sums_chi
+  available_tests['seg_sum']['ks_test'] = sU.compare_sums_ks
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    # If True the metric needs to be able to handle a passed in Distribution
+    self.acceptsDistribution = True
+    self.metricType = None
+
+  def handleInput(self, paramInput):
+    """
+      Method that reads the portion of the xml input that belongs to this specialized class
+      and initialize internal parameters
+      @ In, paramInput, InputData.parameterInput, input specs
+      @ Out, None
+    """
+    self.distParams = {}
+    for child in paramInput.subparts:
+      if child.getName() == "metricType":
+        self.metricType = list(elem.strip() for elem in child.value.split('|'))
+      else:
+        self.distParams[child.getName()] = child.value
+    if self.metricType[0] not in self.__class__.availMetrics.keys() or self.metricType[1] not in self.__class__.availMetrics[self.metricType[0]].keys():
+      self.raiseAnError(IOError, "Metric '", self.name, "' with metricType '", self.metricType[0], "|", self.metricType[1], "' is not valid!")
+
+  def run(self, x, y, weights=None, axis=0, **kwargs):
+    """
+      This method computes difference between two points x and y based on given metric
+      @ In, x,  instance of Distributions.Distribution, tuple or list, array containing data of x,
+        or given distribution.
+      @ In, y, instance of Distributions.Distribution, tuple or list, array containing data of y,
+        or given distribution.
+      @ In, weights, array_like (numpy.ndarray or list), optional, not used in this metric
+      @ In, axis, integer, optional, default is 0, not used for this metric.
+      @ In, kwargs,dict, dictionary of parameters characteristic of each metric
+      @ Out, value, float, metric result, CDF area difference
+    """
+    if isinstance(x,np.ndarray) and isinstance(y,np.ndarray):
+      assert x.shape == y.shape, "Input data x, y should have the same shape!"
```

Comment:
Note that asserts are skipped when run in default RAVEN mode (python's opt mode). Should this be a `self.raiseAnError` or is this assertion just a developer check?